backend/graphql/types/courseType.ts

```diff
@@ -24,10 +24,10 @@ const courseType = gql`
   type CourseResponseDTO {
     id: ID!
     title: String!
-    description: String
-    image: String
-    previewImage: String
-    modules: [ModuleDTO]
+    description: String!
+    image: String!
+    previewImage: String!
+    modules: [ModuleResponseDTO]!
```

Comment:
Why not `[ModuleResponseDTO!]!` -- ie. why no strict type check on `ModuleResponseDTO`, do we not expect that the response will always be of type `ModuleResponseDTO`? If it's because of missing ID, I'd rather we make id optional on the other DTO object

Reply:
Honestly I just didn't know that you needed to do [ModuleResponseDTO!]!